lib/amazon_ssa_support/miq_ec2_vm/miq_ec2_vm.rb

```diff
@@ -3,20 +3,20 @@
 
 module AmazonSsaSupport
   class MiqEC2Vm
-    def self.new(ec2_id, host_instance, ec2)
+    def self.new(ec2_id, host_instance, ec2, ost = nil)
+      ec2_obj = nil
+
       if ec2_id[0, 4] == "ami-"
         ec2_obj = ec2.image(ec2_id)
-        raise "MiqEC2Vm.new: EC2 Image #{ec2_id} not found" unless ec2_obj
-        raise "MiqEC2Vm.new: Image of instance store isn't supported!" unless ec2_obj.root_device_type == "ebs"
-        MiqEC2EbsImage.new(ec2_obj, host_instance, ec2)
       elsif ec2_id[0, 2] == "i-"
         ec2_obj = ec2.instance(ec2_id)
-        raise "MiqEC2Vm.new: EC2 Instance #{ec2_id} not found" unless ec2_obj
-        raise "MiqEC2Vm.new: Instance of instance store isn't supported!" unless ec2_obj.root_device_type == "ebs"
-        MiqEC2EbsInstance.new(ec2_obj, host_instance, ec2)
       else
         raise "MiqEC2Vm.new: unrecognized ec2 ID #{ec2_id}"
       end
+
+      raise "MiqEC2Vm.new: Instance store isn't supported!" unless ec2_obj&.root_device_type == "ebs"
+
+      ec2_obj.respond_to?(:instance_id) ? MiqEC2EbsInstance.new(ec2_obj, host_instance, ec2, ost) : MiqEC2EbsImage.new(ec2_obj, host_instance, ec2, ost)
```

Comment:
This conditional seems strange given we've already performed essentially the same check above.
The refactored `raise unless == "ebs"` no longer indicates if the target is an instance or image.
This no longer indicates the offending `ec2_id` if the object isn't found.

Overall, the code was probably better the way it was.

Reply:
The old codes were marked as duplicate by codeclimate checking. I'll change its setting and let it ignore this part.